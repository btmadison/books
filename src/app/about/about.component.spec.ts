import { Spectator, createComponentFactory } from '@ngneat/spectator/jest';
import { NO_ERRORS_SCHEMA } from '@angular/core';

import { AboutComponent } from './about.component';
import { HighlightDirective } from '../shared/highlight.directive';


// [8] testing directives, very similar to testing components [1]
// use NO_ERRORS_SCHEMA (8.1)
// you may need to cast selectors as HTMLElement, instead of the default Element (8.2)

describe('[7] Testing directives: very similar to components', () => {
  let component: AboutComponent;
  let spectator: Spectator<AboutComponent>;

  const createComponent = createComponentFactory({
    component: AboutComponent,
    declarations: [HighlightDirective],
    schemas: [ NO_ERRORS_SCHEMA ] // use NO_ERRORS_SCHEMA (8.1)
  });

  beforeEach(() => {
    spectator = createComponent();
    component = spectator.component;
  });

  it('Use NO_ERRORS_SCHEMA and may need to cast selectors as HTMLElement', () => {
    // (8.2) IMPORTANT: you may need to cast selectors as HTMLElement, instead of the default Element
    const h2: HTMLElement = spectator.query('h2');
    expect(h2).toBeTruthy();
    expect(h2.style.backgroundColor).toBe('skyblue');
  });

});
