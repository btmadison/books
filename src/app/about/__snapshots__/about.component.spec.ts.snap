// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[7] Testing directives: very similar to components Use NO_ERRORS_SCHEMA and may need to cast selectors as HTMLElement 1`] = `
<ng-component
  __ngContext__={[Function LRootView]}
>
  <h2
    highlight="skyblue"
    style="background-color: skyblue;"
  >
    About
  </h2><h3>
    Quote of the day:
  </h3><twain-quote />
</ng-component>
`;
